pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix

```diff
@@ -31,9 +31,17 @@ buildDunePackage rec {
   useDune2 = true;
   minimumOCamlVersion = "4.06";
 
-  buildInputs = [ yojson stdlib-shims ocaml-syntax-shims ];
+  buildInputs =
+    if lib.versionAtLeast version "1.7.0" then
+      [ ]
```

Comment:
Why not use lib.optional?

Reply:
In my opinion, it's harder for a human to read the list of dependency when one also has to evaluate inline expressions.